src/odemis/gui/cont/streams.py

```diff
@@ -1960,10 +1960,10 @@ def _onStreamUpdate(self, stream, updated):
         # current view?)
 
         # Don't mess too much with the spot stream => just copy "should_update"
-        if hasattr(self._tab_data_model, "spotStream"):
-            if stream is self._tab_data_model.spotStream:
-                stream.is_active.value = updated
-                return
+        spots = getattr(self._tab_data_model, "spotStream", None)
+        if stream is spots:
+            stream.is_active.value = updated
+            return
```

Comment:
Maybe a comment similar to the title of the pr, like "stream is still updated if there is no spotStream" would make it a bit clearer to me :)